src/bootloaders/shim-systemd.c

```diff
@@ -0,0 +1,254 @@
+static int  shim_systemd_get_capabilities(const BootManager *manager);
+
+__cbm_export__ const BootLoader shim_systemd_bootloader = {
+        .name = "systemd",
+        .init = shim_systemd_init,
+        .get_kernel_dst = shim_systemd_get_kernel_dst,
+        .install_kernel = shim_systemd_install_kernel,
+        .remove_kernel = shim_systemd_remove_kernel,
+        .set_default_kernel = shim_systemd_set_default_kernel,
+        .needs_install = shim_systemd_needs_install,
+        .needs_update = shim_systemd_needs_update,
+        .install = shim_systemd_install,
+        .update = shim_systemd_update,
+        .remove = shim_systemd_remove,
+        .destroy = shim_systemd_destroy,
+        .get_capabilities = shim_systemd_get_capabilities
+};
+
+#if UINTPTR_MAX == 0xffffffffffffffff
+#       define EFI_SUFFIX       "x64.efi"
+#else
+#       define EFI_SUFFIX       "ia32.efi"
+#endif
+
+/* Layout entries, see the layout description at the top of the file. */
+#define SHIM_SRC_DIR            "usr/lib/shim"
+#define SHIM_SRC                SHIM_SRC_DIR "/" "shim" EFI_SUFFIX
+#define MM_SRC                  SHIM_SRC_DIR "/" "mm" EFI_SUFFIX
+#define FB_SRC                  SHIM_SRC_DIR "/" "fb" EFI_SUFFIX
+#define SYSTEMD_SRC_DIR         "usr/lib/systemd/boot/efi"
+#define SYSTEMD_SRC             SYSTEMD_SRC_DIR "/" "systemd-boot" EFI_SUFFIX
+#define DST_DIR                 "/" KERNEL_NAMESPACE
+#define SHIM_DST                DST_DIR "/" "bootloader" EFI_SUFFIX
+#define SYSTEMD_DST             DST_DIR "/" "loader" EFI_SUFFIX
+#define KERNEL_DST_DIR          DST_DIR "/kernel"
+#define SYSTEMD_CONFIG_DIR      "/loader"
+#define SYSTEMD_CONFIG          SYSTEMD_CONFIG_DIR "/loader.conf"
+#define SYSTEMD_ENTRIES         SYSTEMD_CONFIG_DIR "/entries"
+
+static char *shim_src;
+static char *shim_dst_host; /* as accessible by the CMB for file ops. */
+static char *shim_dst_esp; /* absolute location of shim on the ESP. */
+static char *systemd_src;
+static char *systemd_dst_host;
+
+static char *shim_systemd_get_kernel_dst(const BootManager *manager) {
+        (void)manager;
+        return strdup(KERNEL_DST_DIR);
+}
+
+static bool shim_systemd_install_kernel(const BootManager *manager, const Kernel *kernel) {
+        return sd_class_install_kernel_impl(manager, kernel, shim_systemd_get_kernel_dst, NULL);
+}
+
+static bool shim_systemd_remove_kernel(const BootManager *manager, const Kernel *kernel) {
+        return sd_class_remove_kernel(manager, kernel);
+}
+
+static bool shim_systemd_set_default_kernel(const BootManager *manager, const Kernel *kernel) {
+        /* this writes systemd config. systemd has the configuration paths
+         * hardcoded, hence whatever sd_class is doing is OK. */
+        return sd_class_set_default_kernel_impl(manager, kernel, NULL);
+}
+
+static bool exists_identical(const char *path, const char *spath) {
+        if (!nc_file_exists(path)) return false;
```

Comment:
style comment: No matter what, after the condition I want to see a newline. Either with or without braces (I think this project uses braces after every `if`) but newline required.

Reply:
Obviously this comment applies to all the occurrences below as well.

Reply:
> No matter what, after the condition I want to see a newline.
[C substance](https://github.com/btrask/stronglink/blob/master/SUBSTANCE.md) (which is recommended guideline) and I disagree with you. :)

Reply:
Hrm I disagree a little with that guide (which actually has nice things in it), I like to have different lines for the condition and the conditional statement because it prevents me assuming that I changed both like I thought when I see the one line diff.

Reply:
It's much easier to observe patterns when split out over 2 lines, the style guide is plainly wrong here IMHO.

Reply:
OK. New line with braces then. No braces are only good for 1-liners...